# eppi
Pipeline for Classification of Viral Metagenomes

- This pipeline is optimized to work with high complexity NGS-reads such as environmental samples.
- Both reads and contigs were annotated using BLAST, Centrifuge, HMMER, Mash and a neural net
- This pipeline is in beta status yet and comes without any warranty!

## Requirements
- [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
- [FASTX-Toolkit](http://hannonlab.cshl.edu/fastx_toolkit/commandline.html)
- [VrAP](http://www.rna.uni-jena.de/en/vrap/)
- [MetaQUAST](http://bioinf.spbau.ru/metaquast)
- [HISAT2](https://ccb.jhu.edu/software/hisat2/)
- [DIAMOND](https://github.com/bbuchfink/diamond)
- [Centrifuge](https://ccb.jhu.edu/software/centrifuge/manual.shtml)
- [Blast+](https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&DOC_TYPE=Download)
- [HMMER](http://hmmer.org/)
- [Mash](https://github.com/marbl/Mash)
- [SAMtools](http://samtools.sourceforge.net/)
- [Python3](https://www.python.org/)
- [Keras](https://keras.io/)
- [TensorFlow](https://www.tensorflow.org/)
